networkx/algorithms/coloring/greedy_coloring.py

```diff
@@ -360,3 +360,193 @@ def greedy_color(G, strategy="largest_first", interchange=False):
         # Assign the new color to the current node.
         colors[u] = color
     return colors
+
+
+# Tools for coloring with interchanges
+class Node:
```

Comment:
Since the `Node` and `AdjEntry` classes are top-level objects in the module, should we preface them with `_` to indicate they are only intended for internal use?

Reply:
Good idea -- both making the classes private and checking for internal `project` calls.